packages/viewer/src/interactions/interactionApi.ts

```diff
@@ -194,6 +202,70 @@ export class InteractionApi {
     });
   }
 
+  public async rotateCameraAtPoint(
+    delta: Point.Point,
+    starting: Point.Point,
+    depth?: number
+  ): Promise<void> {
+    return this.transformCamera((camera, viewport) => {
+      this.startingCameraInverseProjection =
+        this.startingCameraInverseProjection ||
+        camera.inverseProjectionMatrix();
+      this.startingCameraInverseView =
+        this.startingCameraInverseView || camera.inverseViewMatrix();
+
+      if (
+        this.startingCameraInverseProjection != null &&
+        this.startingCameraInverseView != null &&
+        depth != null
+      ) {
+        const upVector = Vector3.normalize(camera.up);
+        const lookAt = Vector3.normalize(
+          Vector3.subtract(camera.lookAt, camera.position)
+        );
+
+        const crossX = Vector3.cross(upVector, lookAt);
+        const crossY = Vector3.cross(lookAt, crossX);
+
+        const mouseToWorld = Vector3.normalize({
+          x: delta.x * crossX.x + delta.y * crossY.x,
+          y: delta.x * crossX.y + delta.y * crossY.y,
+          z: delta.x * crossX.z + delta.y * crossY.z,
+        });
+
+        const normalizedScreenPoint = Vector3.create(
+          (starting.x * 2.0) / viewport.width - 1,
+          1 - (starting.y * 2.0) / viewport.height,
+          depth
+        );
+
+        const point = Matrix4.multiplyVector3(
+          this.startingCameraInverseProjection,
+          normalizedScreenPoint
+        );
+
+        const scaledProjPoint = Vector3.scale(1.0 / point.w, point);
+
+        const viewPoint = Matrix4.multiplyVector3(
+          this.startingCameraInverseView,
+          scaledProjPoint
+        );
```

Comment:
Conversion of screen click to world position. This calculation follows examples from http://bookofhook.com/mousepick.pdf and loosely from https://www.khronos.org/opengl/wiki/GluProject_and_gluUnProject_code (this uses the combined view/projection matrix, and results in the same computed point).

Reply:
One additional note, a couple of these values are adjusted based on testing -- the `y` component of the normalized device coordinate is flipped to go [-1, 1] from [bottom, top]. Additionally, the `z` component of the normalized device coordinate is in the range [0, 1] instead of [-1, 1] based on experimenting with the range, since it resulted in values that were closer to the expected world position